guvnor-webapp/src/main/java/org/guvnor/client/perspectives/M2RepoPerspective.java

```diff
@@ -67,17 +67,18 @@ public Menus getMenus() {
                 .respondsWith( new Command() {
                     @Override
                     public void execute() {
-                        final UploadForm uploadForm = iocManager.lookupBean( UploadForm.class ).getInstance();
-                        //When pop-up is closed destroy bean to avoid memory leak
-                        uploadForm.addCloseHandler( new CloseHandler<PopupPanel>() {
+                        final UploadFormPresenter presenter = iocManager.lookupBean( UploadFormPresenter.class ).getInstance();
+                        final UploadFormViewImpl view = iocManager.lookupBean( UploadFormViewImpl.class ).getInstance();
```

Comment:
New instances of the View will be created "on demand". Therefore the View instance ```@Inject```'ed into the Presenter will be a different instance to that you create here. You should really add a method to the Presenter to add a ```HideHandler``` to the View and *only* instantiate and destroy the Presenter here.